manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java

```diff
@@ -85,40 +86,46 @@ public Bridge getBridge(String id) {
     }
 
     @Transactional
-    public Bridge getReadyBridge(String bridgeId, String customerId) {
-        Bridge bridge = getBridge(bridgeId, customerId);
-        if (BridgeStatus.READY != bridge.getStatus()) {
-            throw new BridgeLifecycleException(String.format("Bridge with id '%s' for customer '%s' is not in the '%s' state.", bridge.getId(), bridge.getCustomerId(), BridgeStatus.READY));
-        }
-        return bridge;
+    public boolean isBridgeReady(Bridge bridge) {
+        return BridgeStatus.READY == bridge.getStatus();
     }
 
-    private Bridge findByIdAndCustomerId(String id, String customerId) {
-        Bridge bridge = bridgeDAO.findByIdAndCustomerId(id, customerId);
-        if (bridge == null) {
-            throw new ItemNotFoundException(String.format("Bridge with id '%s' for customer '%s' does not exist", id, customerId));
+    @Override
+    public Bridge getBridgeByBridgeIdentifier(String bridgeIdentifier, String customerId) {
+        Optional<Bridge> bridgeOptional = getBridgeByIdAndCustomerId(bridgeIdentifier, customerId);
+        if (bridgeOptional.isPresent()) {
+            return bridgeOptional.get();
+        }
+
+        bridgeOptional = getBridgeByNameAndCustomerId(bridgeIdentifier, customerId);
+        if (bridgeOptional.isPresent()) {
+            return bridgeOptional.get();
```

Comment:
I think we should create a unified query with a condition like `WHERE customer_id = <customerId> AND (id = <bridgeIdentifier> OR name = <bridgeIdentifier>)`. What do you think?

Reply:
done